pytket/phir/phirgen.py

```diff
@@ -357,11 +357,13 @@ def extract_wasm_args_and_returns(
 ) -> tuple[list[str], list[str]]:
     """Extract the wasm args and return values as whole register names."""
     # This slice removes the extra `_w` cregs (wires) that are not part of the
-    # circuit, and the output args which are appended after the input args
+    # circuit and the output args, which are appended after the input args
     slice_index = op.num_w + sum(op.output_widths)
     only_args = command.args[:-slice_index]
+    # Eliminate conditional bits from the front of the args
+    input_args = only_args[len(only_args) - op.n_inputs :]
```

Comment:
Not sure why the condition bits appear in the args.

Reply:
Same, to me throughout the project some of the design choices in the pytket API have been quite puzzling, especially those that are undocumented, of course.

Reply:
Looks like condition bits do appear in the args of all Conditional commands (not just those whose conditioned op is WASM). This seems reasonable, but are we making a similar adjustment for non-WASM conditionals? I suppose we must be -- just couldn't see it in the code.

Reply:
I think the difference in other conditionals is that the arguments are qubits are more easily accessible through `command.qubits` property, rather than trying to derive from the long list of `args`. Though you are right that we may have bugs lurking where it might be necessary to apply non-wasm classical operations conditionally.

I am going to take a look.

Reply:
Okay, at least for SetBits and other classical ops, we are using `command.bits` which is again safe. I'll keep a watch for other classical operations as I add them. Added a test as well.